ocaml/xapi/xapi_pool.ml

```diff
@@ -807,6 +807,37 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
               (pool_joining_host_ca_certificates_conflict, !conflicting_names)
           )
   in
+  let assert_no_host_pending_mandatory_guidance () =
+    (* Assert that there is no host pending mandatory guidance on the joiner or
+       the remote pool coordinator.
+    *)
+    Repository_helpers.assert_no_host_pending_mandatory_guidance ~__context
```

Comment:
Could the check on the joiner be extended to the recommended and full as well? The toolstack on the joiner will be restarted after joining. If there is a pending Toolstack restart (even as a full guidance), it probably means that a newer toolstack (than the one on the coordinator) on the joiner will start after joining.
But usually this is a rare case.

Reply:
I think only mandatory guidance is checked is because now RestartToolstack guidance only set as mandatory guidance.
@robhoes Is it possible that RestartToolstack guidance would be set as recommended or full guidance in updates so that we will need to cover all type of guidance here?

Reply:
I see it this way: any guidance that must be followed immediately to avoid things breaking must be part of the "mandatory guidance". This includes breaking pool join. And yes, usually we make restarting the Toolstack mandatory if the Toolstack is updated.

Reply:
> usually we make restarting the Toolstack mandatory if the Toolstack is updated.

Yeah, in this case, it would be fine to check the mandatory guidance only.